### ISSUE TITLE
[question] conan search: non-default options

### ISSUE THREAD

**Issue by wpalfi** [conan-io/conan#6652]
How can I figure out the non-default options when searching for a binary package?

E.g I want to use opencv:
```
conan search opencv/4.1.1@conan/stable -r conan-center -q "compiler.version=15 AND build_type=Release AND arch=x86_64 AND shared=False"
```
It has ~20 options. But there is no binary package with default options. So how can I know which of them I need to set manually?

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).

**Comment by memsharded** [comment]
Hi @wpalfi 

Sorry, I don't fully understand the question. The above command returns 4 different binaries that match that configuration. Could you please clarify your question? Thanks

**Comment by wpalfi** [comment]
The different binaries have a lot of options. I would like to install one of them. I can manually set all these options 
```
conan install opencv/4.1.1@conan/stable -o carotene=False -o contrib=False -o ...(20 more)...
```
but I would prefer setting only a minimum of non-default options. How can I find out which?

**Comment by memsharded** [comment]
Hi,

I am afraid I still don't get it. You don't need to specify all the options, only the ones that you want a specific value. The others will automatically use the default options. Don't need to define the 20 of them.

**Comment by wpalfi** [comment]
Hm. My question is: I dont care about all the options. I just want to find a (minimal) set of options that leads to an existing binary package. I can do that by hand but it is tedious E.g. it would be nice if `conan search` could highlight these non-default options. 
Btw. The problem with opencv is that the binary packages are outdated from recipe.

**Comment by memsharded** [comment]
``opencv/4.1.1@conan/stable`` is the old package, we are moving to a new way of creating packages with the ``conan-center-index`` repository. Those packages don't have ``user/channel`` like ``conan/stable`` anymore. Yes, it is possible that such package is no longer maintain.

Your use case, still not clear to me sorry. Lets say that this is the current output of ``conan search``:

```bash
conan search zlib/1.2.11@
Existing packages for recipe zlib/1.2.11:

    Package_ID: 6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7
        [options]
            minizip: False
            shared: False
        [settings]
            arch: x86_64
            build_type: Release
            compiler: Visual Studio
            compiler.runtime: MD
            compiler.version: 15
            os: Windows
        Outdated from recipe: False
```

What is the output that you will like to see instead of that one?

**Comment by wpalfi** [comment]
Something like this:

```
>conan search --nondefault zlib/1.2.11@
Existing packages for recipe zlib/1.2.11:

    Package_ID: 6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7

    Package_ID: 8cf01e2f50fcd6b63525e70584df0326550364e1
        [settings]
            build_type: Debug
            compiler.runtime: MDd

    Package_ID: 970e773c5651dc2560f86200a4ea56c23f568ff9
        [options]
            shared: True

    Package_ID: ff82a1e70ba8430648a79986385b20a3648f8c19
        [options]
            shared: True
        [settings]
            build_type: Debug
            compiler.runtime: MDd

>conan install zlib/1.2.11@
   => 6cc50b139b9c3d27b3e9042d5f5372d327b3a9f7

>conan install zlib/1.2.11@ -s build_type=Debug
   => 8cf01e2f50fcd6b63525e70584df0326550364e1

>conan install zlib/1.2.11@ -o shared=True
   => 970e773c5651dc2560f86200a4ea56c23f568ff9
   
>conan install zlib/1.2.11@ -o shared=True -s build_type=Debug
   => ff82a1e70ba8430648a79986385b20a3648f8c19
```

Looks trivial here of course, but that would help a lot when you have many options and binary packages.

**Comment by memsharded** [comment]
This will be included in Conan 2.2 with the ``conan list --profile-filter`` arguments
